integration/virtual_kubelet.go

```diff
@@ -11,19 +11,64 @@ import (
 	"github.com/kubernetes-sig-testing/frameworks/integration/internal"
 )
 
+// VirtualKubelet knows how to run a virtual-kubelet
 type VirtualKubelet struct {
-	Path         string
-	Args         []string
-	ConfDir      string
+	// Path is the path to the virtual-kubelet binary.
+	//
+	// If this is left as the empty string, we will attempt to locate a binary,
+	// by checking for the TEST_ASSET_VIRTUAL_KUBELET environment variable, and
+	// the default test assets directory. See the "Binaries" section above (in
+	// doc.go) for details.
+	Path string
+
+	// Args is a list of arguments which will passed to the virtual-kubelet
+	// binary.  Before they are passed on, they will be evaluated as go-template
+	// strings.
+	// This means you can use fields which are defined and exported on this
+	// VirtualKubelet struct (e.g. "--kubeconfig={{ .ConfDir }}/kube.conf").
+	// Those templates will be evaluated after the defaulting of the
+	// VirtualKubelet's fields has already happened and just before the binary
+	// actually gets started. Thus you have access to caluclated fields like
+	// `ConfDir` and others.
+	//
+	// If not specified, the minimal set of arguments to run the APIServer will
+	// be used.
+	Args []string
+
+	// ConfDir is a path to a directory containing whatever configuration the
+	// virtual kubelet needs.
+	//
+	// If left unspecified, then the Start() method will create a fresh temporary
+	// directory, and the Stop() method will clean it up.
+	ConfDir string
+
+	// APIServerURL is the URL pointing to the APIServer of the control plane
+	// this VirtualKubelet should connect to.
+	//
+	// It can be set directly on the struct, or indirectly by calling the
+	// `RegisterTo` method.
+	// This setting is mandatory and an error will be emitted when this is not
+	// set at start time.
 	APIServerURL *url.URL
+
+	// StartTimeout, StopTimeout specify the time the APIServer is allowed to
```

Comment:
Not apiserver, here and below :-)

Reply:
Thanks. I will recheck on the docs and typos overall.